streams/src/main/java/org/apache/kafka/streams/processor/internals/AssignedTasks.java

```diff
@@ -0,0 +1,459 @@
+    }
+
+    void commit() {
+        performTimedAction(commitAction);
+    }
+
+    void maybeCommit() {
+        performTimedAction(maybeCommitAction);
+    }
+
+    private void performTimedAction(final TaskAction action) {
+        timerStartedMs = time.milliseconds();
+        final RuntimeException exception = applyToRunningTasks(action, false);
+        if (exception != null) {
+            throw exception;
+        }
+    }
+
+    int process() {
+        timerStartedMs = time.milliseconds();
+        int processed = 0;
+        timerStartedMs = time.milliseconds();
+        for (final Task task : running()) {
+            if (task.process()) {
+                processSensor.record(computeLatency(), timerStartedMs);
+                processed++;
+                try {
+                    if (task.maybePunctuateStreamTime()) {
+                        punctuateSensor.record(computeLatency(), timerStartedMs);
+                    }
+                } catch (final KafkaException e) {
+                    log.error("{} Failed to punctuate active task {} due to the following error:", logPrefix, task.id(), e);
+                    throw e;
+                }
+            }
+        }
+        return processed;
+    }
+
+    RuntimeException applyToRunningTasks(final TaskAction action, final boolean throwException) {
+        RuntimeException firstException = null;
+
+        for (Iterator<Task> it = running().iterator(); it.hasNext(); ) {
+            final Task task = it.next();
+            try {
+                action.apply(task);
+            } catch (final CommitFailedException e) {
+                // commit failed. This is already logged inside the task as WARN and we can just log it again here.
+                log.warn("{} Failed to commit {} {} during {} state due to CommitFailedException; this task may be no longer owned by the thread", logPrefix, taskTypeName, task.id(), action.name());
+            } catch (final ProducerFencedException e) {
+                closeZombieTask(task);
+                it.remove();
+            } catch (final RuntimeException t) {
+                log.error("{} Failed to {} {} {} due to the following error:",
+                          logPrefix,
+                          action.name(),
+                          taskTypeName,
+                          task.id(),
+                          t);
+                if (throwException) {
+                    throw t;
+                }
+
+                if (firstException == null) {
+                    firstException = t;
+                }
+            }
+        }
+
+        return firstException;
+    }
+
+    void closeNonAssignedSuspendedTasks(final Map<TaskId, Set<TopicPartition>> newAssignment) {
+        closeNonAssignedTask(newAssignment, suspended.values().iterator());
+    }
+
+    private void closeNonAssignedTask(final Map<TaskId, Set<TopicPartition>> newAssignment, final Iterator<Task> standByTaskIterator) {
```

Comment:
left over from a previous incantation where i was suspending restoring tasks, too. Thanks for pointing out.